File: Binary Tree Inorder Traversal.cpp
#include <iostream>
#include <vector>
#include <stack>

using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

vector<int> inorderTraversal(TreeNode* root) {
    vector<int> result;
    stack<TreeNode*> stack;
    TreeNode* current = root;

    while (current != NULL || !stack.empty()) {
        while (current != NULL) {
            stack.push(current);
            current = current->left;
        }
        current = stack.top();
        stack.pop();
        result.push_back(current->val);
        current = current->right;
    }

    return result;
}

int main() {
    // Example usage:
    TreeNode* root = new TreeNode(1);
    root->right = new TreeNode(2);
    root->right->left = new TreeNode(3);

    vector<int> result = inorderTraversal(root);
    for (int val : result) {
        cout << val << " ";
    }

    return 0;
}

File: Binary Tree Inorder Traversal.cs
/++
* Definition for a binary tree node.
* public class TreeNode {
*     public class in val;
*     public TreeNode left;
*     public TreeNode right;
*     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
*       this. val = val;
*       this. left = left;
*       this. right = right;
*    }
* }
*/
public class solution {
    public Ilist<int> InorderTraversal(TreeNode) {
        Ilist<int> result=new list<int>();

        TreeNode currentNode=root;

        if(currentNode!=null){
            foreach(int r in InorderTraversal(currentNode.left))
            result .Add(r);
            result .Add(currentNode. val);
            foreach(int r in InorderTraversal(currentNode.right))
            result .Add(r);
        }

        return result;
    }
}